netbox/circuits/models/virtual_circuits.py

```diff
@@ -0,0 +1,164 @@
+from functools import cached_property
+
+from django.core.exceptions import ValidationError
+from django.db import models
+from django.urls import reverse
+from django.utils.translation import gettext_lazy as _
+
+from circuits.choices import *
+from netbox.models import ChangeLoggedModel, PrimaryModel
+from netbox.models.features import CustomFieldsMixin, CustomLinksMixin, TagsMixin
+
+__all__ = (
+    'VirtualCircuit',
+    'VirtualCircuitTermination',
+)
+
+
+class VirtualCircuit(PrimaryModel):
+    """
+    A virtual connection between two or more endpoints, delivered across one or more physical circuits.
+    """
+    cid = models.CharField(
+        max_length=100,
+        verbose_name=_('circuit ID'),
+        help_text=_('Unique circuit ID')
+    )
+    provider_network = models.ForeignKey(
```

Comment:
Just for my own understanding, virtual circuits don't appear to be very congruent to regular circuits in terms of the data model -- the latter have `provider_network` as part of the CircuitTermination, whereas here they're a top-level attribute of the VirtualCircuit rather than the termination. I assume that's an intentional part of the design.

Reply:
Right; this is because a physical circuit provides access from a customer POP _to_ a provider network, whereas a virtual circuit provides connectivity among multiple points _across_ a provider network.